src/storage/store/string_column.cpp

```diff
@@ -25,12 +25,36 @@ StringColumn::StringColumn(std::string name, LogicalType dataType,
     : Column{name, std::move(dataType), metaDAHeaderInfo, dataFH, metadataDAC, bufferManager, wal,
           transaction, enableCompression, true /* requireNullColumn */},
       dictionary{name, metaDAHeaderInfo, dataFH, metadataDAC, bufferManager, wal, transaction,
-          enableCompression} {}
+          enableCompression} {
+
+    // add child column for index
+    auto indexColName = StorageUtils::getColumnName(name, StorageUtils::ColumnType::INDEX, "index");
+    indexColumn = ColumnFactory::createColumn(indexColName, *LogicalType::UINT32(),
+        *metaDAHeaderInfo.childrenInfos.at(2), dataFH, metadataDAC, bufferManager, wal, transaction,
+        enableCompression);
+}
+
+Column::ChunkState& StringColumn::getIndexState(ChunkState& state) {
+    static constexpr size_t indexColumnChildIndex =
+        static_cast<size_t>(StringChunkData::ChildType::INDEX);
+    KU_ASSERT(state.childrenStates.size() > indexColumnChildIndex);
+    return state.childrenStates[indexColumnChildIndex];
+}
+
+const Column::ChunkState& StringColumn::getIndexState(const ChunkState& state) {
+    static constexpr size_t indexColumnChildIndex =
+        static_cast<size_t>(StringChunkData::ChildType::INDEX);
+    KU_ASSERT(state.childrenStates.size() > indexColumnChildIndex);
+    return state.childrenStates[indexColumnChildIndex];
+}
 
 void StringColumn::initChunkState(Transaction* transaction, node_group_idx_t nodeGroupIdx,
     ChunkState& state) {
     Column::initChunkState(transaction, nodeGroupIdx, state);
-    dictionary.initChunkState(transaction, nodeGroupIdx, state);
+    state.childrenStates.resize(CHILD_COLUMN_COUNT);
+    indexColumn->initChunkState(transaction, nodeGroupIdx, getIndexState(state));
+    dictionary.initChunkState(transaction, nodeGroupIdx, getIndexState(state));
```

Comment:
Currently, the hierarchy for ChunkStates is Main > Index > Dictionary(Offset, Data). So there are 3 levels. Let me know if we want to change this to 2 levels (something like Main > Index, Dictionary)

Reply:
Shouldn't be too bad, one potentially painful thing would be that the DictionaryColumn hard-codes the indices at which it references the child states which doesn't mix too well as-is with setting child state indices for columns outside the DictionaryColumn (in this case the index column). I'll try to restructure things so that they fit together nicer while following the MetadataDAHInfo structure.